### ISSUE TITLE
Summary statistic files contain NaN values

### ISSUE THREAD

**Issue by SBNeuro1** [zhwm/SharePro_coloc#1]
Hi Wenmin,

Thanks very much for making this package available it's really helping me answer some tough questions. It's also been very easy to set up and use. I have managed to analyse almost 150 genomic regions of interest to me across multiple phenotypes, but there is one phenotype that is giving me trouble. I keep getting an error: 

Error output:
Z:\SharePro_coloc\src\SharePro\sharepro_coloc.py:144: RuntimeWarning: invalid value encountered in divide
  z = beta / se
Traceback (most recent call last):
  File "Z:\SharePro_coloc\src\SharePro\sharepro_coloc.py", line 206, in <module>
    main(args)
  File "Z:\SharePro_coloc\src\SharePro\sharepro_coloc.py", line 175, in main
    assert not np.isnan(bhatlst).any(), "The summary statistic files contain NaN values"
AssertionError: The summary statistic files contain NaN values

I know that one of the sumstats is fine because I used it across all the SharePro analyses that did work. The issue is that I've analysed the problematic sumstat and I did find some NA and zero values, but I am still getting the same error even after removing these values. Do you have any idea what might be causing this error?

**Comment by zhwm** [comment]
Hi,

Thank you for the note.

I believe there are still NA values in the input data, as indicated by the assertion error. Could you please double check?

Best,
Wenmin

**Comment by SBNeuro1** [comment]
Hi Wenmin,

Thanks very much for your reply. It helped me get around this issue after doing a more thorough search across all my sumstats.
The error was occurring because some of the sumstats required me to calculate missing information, and in an isolated number of loci (which affected only one of my phenotype pairs), there were zero values in columns that I needed to calculate these missing values. I hadn't imposed any filters in these columns, so the missing info columns ended up containing NaN values during SharePro processing.
Once I had removed these zero values, the process went smoothly and I was able to complete my analysis!

Cheers,

SB
